### PR TITLE
Feature/Create JSDocs

### PR DESCRIPTION
**Pull request by Sisha3342**
@exadel-inc/ui-playground-team

### CODE REVIEW

File: src/snippets/snippets.ts [hunk 1]
```diff
@@ -50,6 +57,7 @@ export class UIPSnippets extends UIPPlugin {
     return $scroll;
   }
 
+  /** Render snippets list. */
   protected rerender(): void {
```
Comment (nattallius) [inline src/snippets/snippets.ts:61]: It's just look strange for me to see comment with _render_ to method called _rerender_. At all, i'd better rename this method, cause by it's name i have no idea what it does and what it rerenders

File: src/settings/settings.ts [hunk 2]
```diff
@@ -4,9 +4,17 @@ import {attr} from '@exadel/esl/modules/esl-base-element/core';
...
 
+/**
+ * Container class for [settings]{@link UIPSetting}.
+ * @see {@link UIPPlugin}
+ */
 export class UIPSettings extends UIPPlugin {
   public static is = 'uip-settings';
 
+  /**
+   * Attribute which sets all inner [settings]{@link UIPSetting}
```
Comment (nattallius) [inline src/settings/settings.ts:15]: isn't it a possessive case for _settings'_?

**Review by nattallius — APPROVED** [review]

File: src/core/state-model.ts [hunk 3]
```diff
@@ -53,6 +74,7 @@ export class UIPStateModel extends Observable {
     this.fire();
```
Comment (julia-murashko) [inline src/core/state-model.ts:74]: Could we simplify the code below
```
 if ('transform' in cfg) {
      UIPStateModel.setAttribute(elements, attribute, cfg.transform);
    } else {
      UIPStateModel.setAttribute(elements, attribute, cfg.value);
    }
```
to 
`UIPStateModel.setAttribute(elements, attribute, 'transform' in cfg ? cfg.transform :  cfg.value);`

File: src/core/state-model.ts [hunk 4]
```diff
@@ -53,6 +74,7 @@ export class UIPStateModel extends Observable {
...
   }
 
+  /** Transform attributes values for specified elements. */
```
Comment (julia-murashko) [inline src/core/state-model.ts:77]: just `attribute values`

File: src/editor/editor.ts [hunk 5]
```diff
@@ -10,26 +10,40 @@ import {jsonAttr} from '@exadel/esl/modules/esl-base-element/core';
 
 import {UIPPlugin} from '../core/plugin';
 
+/** Config interface for customizing editor's behaviour. */
 interface EditorConfig {
+  /** Editor's appearance theme. */
   theme: string;
+  /** Text mode used. */
   mode: string;
```
Comment (julia-murashko) [inline src/editor/editor.ts:18]: Text mode is used or used text mode?

**Review by julia-murashko — APPROVED** [review]

File: src/core/state-model.ts [hunk 6]
```diff
@@ -35,24 +54,23 @@ export class UIPStateModel extends Observable {
     return this._lastModifier;
   }
 
+  /** Get attributes values from targets. */
```
Comment (ala-n) [inline src/core/state-model.ts:57]:
```
Get attribute values for the matched set of elements
@param target - CSS sector to define target elements
@parm attr - attribute name
@returns array of matched elements attribute value (uses the element placement order)
```

File: src/core/state-model.ts [hunk 7]
```diff
@@ -35,24 +54,23 @@ export class UIPStateModel extends Observable {
...
-      UIPStateModel.setAttribute(elements, attribute, cfg.transform);
-    } else {
-      UIPStateModel.setAttribute(elements, attribute, cfg.value);
-    }
+    UIPStateModel.setAttribute(elements, attribute, 'transform' in cfg ? cfg.transform : cfg.value);
     this._lastModifier = modifier;
     this.fire();
   }
 
+  /** Transform attributes values. */
```
Comment (ala-n) [inline src/core/state-model.ts:73]: TODO: devitalize doc

File: src/editor/editor.ts [hunk 8]
```diff
@@ -10,26 +10,42 @@ import {jsonAttr} from '@exadel/esl/modules/esl-base-element/core';
 
 import {UIPPlugin} from '../core/plugin';
 
+/** Config interface for customizing editor's behaviour. */
```
Comment (ala-n) [inline src/editor/editor.ts:13]: Just "Config interface to define inner ACE editor settings"

File: src/editor/editor.ts [hunk 9]
```diff
@@ -10,26 +10,42 @@ import {jsonAttr} from '@exadel/esl/modules/esl-base-element/core';
...
 interface EditorConfig {
+  /** Editor's appearance theme. */
   theme: string;
+  /**
+   * Text mode which is used
+   * inside the editor.
+   */
   mode: string;
```
Comment (ala-n) [inline src/editor/editor.ts:21]: Do we need it basically?

File: src/editor/editor.ts [hunk 10]
```diff
@@ -10,26 +10,42 @@ import {jsonAttr} from '@exadel/esl/modules/esl-base-element/core';
...
+  /** Position of the vertical line for wrapping. */
```
Comment (ala-n) [inline src/editor/editor.ts:22]: Cosmetic: link to ACE doc

File: src/editor/editor.ts [hunk 11]
```diff
@@ -10,26 +10,42 @@ import {jsonAttr} from '@exadel/esl/modules/esl-base-element/core';
...
   printMarginColumn: number;
+  /** Limit of characters before wrapping. */
   wrap: number;
 }
 
+/**
+ * Code editor for changing current markup. Allows user to manually configure
```
Comment (ala-n) [inline src/editor/editor.ts:29]:
```
Editor UIPPlugin custom element definition.
Uses ACE UI code editor to provide an ability to modify UIP state markup.
@extends UIPPlugin 
```

File: src/editor/editor.ts [hunk 12]
```diff
@@ -10,26 +10,42 @@ import {jsonAttr} from '@exadel/esl/modules/esl-base-element/core';
...
     theme: 'ace/theme/chrome',
     mode: 'ace/mode/html',
     printMarginColumn: -1,
     wrap: true,
   };
 
+  /** Editor's [config]{@link EditorConfig} passed through attribute. */
   @jsonAttr()
   public editorConfig: EditorConfig;
   protected editor: Ace.Editor;
```
Comment (ala-n) [inline src/editor/editor.ts:46]: `/** Wrapped ACE editor instance */`

File: src/editor/editor.ts [hunk 13]
```diff
@@ -10,26 +10,42 @@ import {jsonAttr} from '@exadel/esl/modules/esl-base-element/core';
...
 
+  /** Merge [default config]{@link defaultOptions} with editor's [config]{@link editorConfig}. */
   protected get mergedEditorConfig(): EditorConfig {
```
Comment (ala-n) [inline src/editor/editor.ts:49]: Note: Regardless of the fact that the getter is a function, it defines the simple object property, so the description should build accordingly.
Also, fill free to use the special word `@returns` it helps to build description text for things that return something

File: src/options/options.ts [hunk 14]
```diff
@@ -123,6 +138,7 @@ export class UIPOptions extends UIPPlugin {
     this.changeEditorTheme(theme);
   }
 
+  /** Apply horizontal mode for mobile breakpoints. */
```
Comment (ala-n) [inline src/options/options.ts:141]:
```
Callback to track resize event.
Applys horizontal mode for mobile breakpoints.
```

File: src/settings/setting/bool-setting/bool-setting.ts [hunk 15]
```diff
@@ -5,14 +5,29 @@ import {ChangeAttrConfig, UIPStateModel} from '../../../core/state-model';
 import TokenListUtils from '../../../utils/token-list/token-list-utils';
 import {WARN} from '../../../utils/warn-messages/warn';
 
+/**
+ * Custom setting for adding/removing attributes or appending values to attribute.
+ * @see {@link UIPSetting}
+ */
 export class UIPBoolSetting extends UIPSetting {
   public static is = 'uip-bool-setting';
+  /** Class added when setting has inconsistent state. */
```
Comment (ala-n) [inline src/settings/setting/bool-setting/bool-setting.ts:14]: Cosmetic "CSS class ..."

File: src/settings/setting/bool-setting/bool-setting.ts [hunk 16]
```diff
@@ -5,14 +5,29 @@ import {ChangeAttrConfig, UIPStateModel} from '../../../core/state-model';
...
+  /** Setting's visible name. */
   @attr({defaultValue: ''}) public label: string;
+  /**
+   * Value used for updating [attribute's]{@link UIPSetting#attribute} value.
+   * If it's unset, setting adds/removes [attribute]{@link UIPSetting#attribute}.
+   */
   @attr({defaultValue: ''}) public value: string;
+  /**
+   * Attribute which sets mode for setting.
+   * Replace - replacing [attribute's]{@link UIPSetting#attribute} value with setting's value.
```
Comment (ala-n) [inline src/settings/setting/bool-setting/bool-setting.ts:26]: Use ` ` ` to define code related quotes like literals
```
`replace` - ...
```

File: src/snippets/snippets.ts [hunk 17]
```diff
@@ -2,10 +2,17 @@ import {memoize} from '@exadel/esl/modules/esl-utils/decorators/memoize';
 import {bind} from '@exadel/esl/modules/esl-utils/decorators/bind';
 import {UIPPlugin} from '../core/plugin';
 
+/**
+ * Container for snippets (component's templates).
```
[the remark this answers is not in the document]
Comment (ala-n) [reply src/snippets/snippets.ts]: Or, basically, 'Container element'

**Review by yadamskaya — APPROVED** [review]

**Review by ala-n — APPROVED** [review]

**Pull request merged**